src/ui/Search.tsx

```diff
@@ -1,30 +1,48 @@
 import React, {Component} from 'react';
-import PropTypes from 'prop-types';
 import Dialog from '../components/Dialog';
 import {Tab, Tabs, TabList, TabPanel} from 'react-tabs';
 import 'react-tabs/style/react-tabs.less';
 import {say, getBeginCursor, getEndCursor, playSound, WRAP} from '../utils';
+import { BlockEditorComponentClass } from './BlockEditor';
+import { Searcher } from './searchers/Searcher';
+import { GetProps } from 'react-redux';
+
+export default function attachSearch(
+  Editor: BlockEditorComponentClass,
+  searchModes: Searcher<any, any>[]
+) {
```

Comment:
Is it true that `attachSearch` should only be called on `BlockEditor`? That's the only usage of `attachSearch` I could find.

Reply:
Yes. If there's a better approach than a HOC, I'm all for it!

Reply:
Well, HOC are primarily for sharing behavior across components. If you are not sharing behavior, you don't strictly need a HOC. All the stuff that `attachSearch` does could just be part of the block editor.

Reply:
Interesting! I thought this was a necessary backflip to take. OK, yeah - that can be safely undone.

Reply:
I think I'd rather leave that for a separate PR.